api/v1/services/CheckInService.js

```diff
@@ -0,0 +1,71 @@
+var CheckitError = require('checkit').Error;
+var _ = require('lodash');
+
+var CheckIn = require('../models/CheckIn');
+var UserService = require('../services/UserService');
+var errors = require('../errors');
+var utils = require('../utils');
+
+
+/**
+ * Finds a CheckIn by User ID
+ * @param {Number} userId id of requested user
+ * @returns {Promise} the resolved CheckIn for the user
+ * @throws {NotFoundError} when the user has no check in
+ */
+module.exports.findCheckInByUserId = function (userId){
+    return CheckIn
+        .findByUserId(userId)
+        .then(function (checkin){
+            if (_.isNull(checkin)) {
+                var message = "A check in record cannot be found for the given user";
+                var source = "userId";
+                throw new errors.NotFoundError(message, source);
+            }
+            return checkin;
+        })
+};
+
+/**
+ * Updates the CheckIn values to request
+ * @param {Obejct} attributes to be updated
+ * @returns {Promise} the resolved CheckIn for the user
+ */
+module.exports.updateCheckIn = function (attributes){
+    return module.exports.findCheckInByUserId(attributes.userId)
+        .then(function(checkin) {
+            var updates = {
+                "swag": attributes.swag || checkin.get('swag'),
+                "location": attributes.location || checkin.get('location')
+            };
+            checkin.set(updates, {patch: true});
+            return checkin.save();
+        });
+};
+
+/**
+ * Creates a CheckIn object for given user with the given attributes
+ * @param {Object} attribute values requested
+ * @returns {Promise} resolving to CheckIn object
+ * @throws {InvalidParameterError} when the user has already checked in
+ */
+module.exports.createCheckIn = function (attributes){
```

Comment:
I'd structure this the same way as we do in the other services. It makes more sense to validate the `attributes` before we start making queries to the db that we might not need if `attributes` is invalid